packages/block-editor/src/components/block-card/index.js

```diff
@@ -43,7 +43,9 @@ function BlockCard( { title, icon, description, blockType, className } ) {
 	}, [] );
 
 	const { selectBlock } = useDispatch( blockEditorStore );
-
+	// Only synced patterns are selectable in the editor so change the title to show the sync status.
```

Comment:
Once we have partial synced patterns as well we probably want to do his further back up the tree. I did look at setting this in `useBlockDisplayInformation` but currently we need to use core-data to get the entity record to get the blocks sync status from `wp_block` but core-data imports are not allowed in the block editor.